xmtp_mls/src/groups/device_sync.rs

```diff
@@ -204,111 +174,105 @@ where
             xmtp_common::yield_().await
         }
         self.sync_init().await?;
+        self.handle.increment_metric(SyncMetric::Init);
 
         while let Some(event) = self.stream.next().await {
             let event = event?;
-            match event {
-                LocalEvents::SyncMessage(msg) => match msg {
-                    SyncMessage::Reply { message_id } => {
+
+            if let LocalEvents::SyncEvent(msg) = event {
+                match msg {
+                    SyncEvent::NewSyncGroupFromWelcome => {
```

Comment:
matching & doing the work all in one function just creates a really long function. WDYT about splitting into parts

Reply:
All for it, this was bothering me a bit too